starfish/pipeline/features/pixels/pixel_spot_detector.py

```diff
@@ -1,27 +1,21 @@
-import collections
-import math
-from typing import Tuple, Optional, List
+from typing import List, Tuple, Union
 
-import numpy as np
 import pandas as pd
-from skimage.measure import regionprops, label
-from skimage.measure._regionprops import _RegionProperties
-from sklearn.neighbors import NearestNeighbors
+import numpy as np
 
-from starfish.munge import melt
-from starfish.io import Stack
+
+from starfish.codebook import Codebook
+from starfish.image import ImageStack
+from starfish.intensity_table import IntensityTable
 from ._base import PixelFinderAlgorithmBase
 
 
 class PixelSpotDetector(PixelFinderAlgorithmBase):
 
-    _MerfishDecoderResults = collections.namedtuple(
-        '_MerfishDecoderResults',
-        ['result', 'decoded_img', 'label_img', 'spot_props'])
-
     def __init__(
-            self, codebook: str, distance_threshold: float=0.5176,
-            magnitude_threshold: int=1, area_threshold: int=2, crop_size: int=40, **kwargs) -> None:
+            self, codebook: Union[str, Codebook], distance_threshold: float=0.5176,
```

Comment:
Would prefer that the constructor is more specific about what it accepts (probably Codebook?) and provide factory methods to call the constructor.

Reply:
I can move the loading of the codebook into the PipelineComponent and remove this flexibility from the class. 👍 

It's a symptom of me implementing this before the paired PipelineComponent, thanks for catching.